lib/msal-core/src/ServerRequestParameters.ts

```diff
@@ -65,8 +65,7 @@ export class ServerRequestParameters {
         this.nonce = CryptoUtils.createNewGuid();
         this.state = state && !StringUtils.isEmpty(state) ?  CryptoUtils.createNewGuid() + "|" + state   : CryptoUtils.createNewGuid();
 
-        // TODO: Change this to user passed vs generated with the new PR
-        this.correlationId = CryptoUtils.createNewGuid();
+        this.correlationId = correlationId || CryptoUtils.createNewGuid();
```

Comment:
What happens if you pass a string that isn't a guid? I think we should validate that the provided string is a valid guid and throw an error.

Reply:
+1

Reply:
e.g. add an `isValidRegex` function to the CryptoUtils file that performs a check with this regex: https://stackoverflow.com/questions/7905929/how-to-test-valid-uuid-guid

Reply:
And then throws a new type of ClientAuth error if it fails.

Reply:
Do we care if it is a guid? Does ests?

Reply:
what if the correlation id is 
134u9324324243 and thats how they correlate?

Reply:
I can check with STS if the requirement for correlation-id is a GUID and add this check.

Reply:
So confirmed from STS that they look for a GUID when they parse for correlation-id. So I will go with @jasonnutter' s suggestion to add a check for the same for the user input.

Reply:
Changes made. Please review.